File: 251115/줄 세우기 2/line-up-students-2.py
class Student:
    def __init__(self, h, w, n):
        self.h = h
        self.w = w
        self.n = n

n = int(input())
students = []

for i in range(n):
    h, w = map(int, input().split())
    students.append(Student(h, w, i + 1))

students.sort(key=lambda x: (x.h, -x.w))

for s in students:
    print(s.h, s.w, s.n)

File: 260225/줄 세우기 2/line-up-students-2.js
class Student {
    constructor (height, weight, index) {
        this.height = height;
        this.weight = weight;
        this.index = index;
    }
}

const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const n = Number(input[0]);

const students = [];
for (let i = 1; i <= n; i++) {
    const [height, weight] = input[i].split(' ').map(Number);
    students[i - 1] = new Student(height, weight, i);
}

students.sort((a, b) => {
    if (a.height === b.height) return b.weight - a.weight;
    return a.height - b.height;
})

for (let i = 0; i < students.length; i++) {
    console.log(`${students[i].height} ${students[i].weight} ${students[i].index}`);
}